src/snowflake/cli/plugins/snowpark/package/anaconda.py

```diff
@@ -1,45 +1,79 @@
 from __future__ import annotations
 
 import logging
-from typing import List
+from typing import Dict, List, Set
 
 import requests
 from click import ClickException
-from packaging.version import parse
+from packaging.requirements import InvalidRequirement
+from packaging.requirements import Requirement as PkgRequirement
+from packaging.version import InvalidVersion, parse
 from requests import HTTPError
-from snowflake.cli.plugins.snowpark.models import Requirement, SplitRequirements
+from snowflake.cli.plugins.snowpark.models import (
+    Requirement,
+    SplitRequirements,
+    WheelMetadata,
+)
 
 log = logging.getLogger(__name__)
 
 
+def _standarize_name(name: str) -> str:
+    return WheelMetadata.to_wheel_name_format(name.lower())
+
+
 class AnacondaChannel:
     snowflake_channel_url: str = (
         "https://repo.anaconda.com/pkgs/snowflake/channeldata.json"
     )
 
-    def __init__(self, packages):
-        self._packages = packages
+    def __init__(self, packages: Dict[str, Set[str]]):
+        """[packages] should be a dictionary mapping package name to set of its available versions"""
+        self._packages = {
+            _standarize_name(package_name): versions
+            for package_name, versions in packages.items()
+        }
```

Comment:
We are performing `for` loop here and `from_snowflake` I think it would be much better to encapsulate this logic in a single place (probably in `from_snowflake`). Using `for` loop instead of comprehensions may be better in such case. It will also make tests much cleaner (although you need to adjust them).

Reply:
Fixed.